### ISSUE TITLE
 only-new-issues flag is not working as expected

### ISSUE THREAD

**Issue by gane5hvarma** [golangci/golangci-lint-action#523]
So i have raised a pr with lint issues like multiline and type check. When this flag `only-new-issues` is added, its not checking the linting errors which are introduced. When removed this flag, the lint error is being caught
Here is my github action file
```
name: golangci-lint
permissions:
  contents: read
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
jobs:
  golangci-lint:
    strategy:
      matrix:
        go-version: [ 1.18.x ]
        os: [ ubuntu-latest ]
    runs-on: ${{ matrix.os }}
    steps:
    - name: Install Go
      uses: actions/setup-go@v3
      with:
        go-version: ${{ matrix.go-version }}
    - name: Checkout code
      uses: actions/checkout@v3
    - name: golangci-lint
      uses: golangci/golangci-lint-action@v3.2.0
      with:
        version: latest
        args: --timeout 5m --out-${NO_FUTURE}format colored-line-number
        only-new-issues: true
        skip-cache: true
```

**Comment by jacobbednarz** [comment]
i don't think `only-new-issues` works with pushes; only pull requests as it cannot compare only the change introduced in the PR. i generally only run `golanglint-ci` on the `pull_request` event to handle this.

**Comment by ldez** [comment]
Fixed by #1029
